x-pack/plugins/fleet/public/applications/fleet/components/settings_flyout/confirm_modal.tsx

```diff
@@ -0,0 +1,185 @@
+  ({ changes, onConfirm, onClose }) => {
+    const hasESChanges = changes.some((change) => change.type === 'elasticsearch');
+    const hasFleetServerChanges = changes.some((change) => change.type === 'fleet_server');
+
+    return (
+      <EuiModal maxWidth="768px" onClose={onClose}>
+        <EuiModalHeader>
+          <EuiModalHeaderTitle>
+            <FormattedMessage
+              id="xpack.fleet.settingsConfirmModal.title"
+              defaultMessage="Apply settings to all agent policies"
+            />
+          </EuiModalHeaderTitle>
+        </EuiModalHeader>
+
+        <EuiModalBody>
+          <EuiCallOut
+            title={
+              <FormattedMessage
+                id="xpack.fleet.settingsConfirmModal.calloutTitle"
+                defaultMessage="This action will update all agent policies and enrolled agents."
+              />
+            }
+            color="warning"
+            iconType="alert"
+          >
+            <EuiText size="s">
+              {hasFleetServerChanges && (
+                <p>
+                  <FormattedMessage
+                    id="xpack.fleet.settingsConfirmModal.fleetServerChangedText"
+                    defaultMessage="If agents are unable to connect at the new {fleetServerUrl}, they will log an error and report an unhealthy status. They will remain on the current agent policy version and check for updates at the old URL until they successfully connect at the new URL"
+                    values={{
+                      fleetServerUrl: (
+                        <b>
+                          <FormattedMessage
+                            id="xpack.fleet.settingsConfirmModal.fleetServerUrl"
+                            defaultMessage="Fleet Server URL"
+                          />
+                        </b>
+                      ),
+                    }}
+                  />
+                </p>
+              )}
+
+              {hasESChanges && (
+                <p>
+                  <FormattedMessage
+                    id="xpack.fleet.settingsConfirmModal.eserverChangedText"
+                    defaultMessage="If agents are unable to connect at the new {elasticsearchUrl}, Fleet Server will report them as healthy but they will be unable to send data to Elasticsearch. This will not update URL that Fleet server itself uses to connect to Elasticsearch; you must manually reenroll it to update the URL"
+                    values={{
+                      elasticsearchUrl: (
+                        <b>
+                          <FormattedMessage
+                            id="xpack.fleet.settingsConfirmModal.elasticsearchUrl"
+                            defaultMessage="Elasticsearch URL"
+                          />
+                        </b>
+                      ),
+                    }}
+                  />
+                </p>
+              )}
+            </EuiText>
+          </EuiCallOut>
+
+          {changes.length > 0 && (
```

Comment:
I just noticed the spacing between `EuiCallout` and `EuiBasicTable` is a bit tight. Can you add an `EuiSpacer size="m"` between them? (should go inside the condition here on line 164)

Reply:
sure 👍